nuklear_console_row.h

```diff
@@ -161,57 +138,54 @@ NK_API struct nk_rect nk_console_row_render(nk_console* console) {
   }
 
   // Consume mouse movement before children have a chance to.
-  int numChildren = (int)cvector_size(row->data.children);
+  int numChildren = (int)cvector_size(console->children);
   struct nk_input* input = &console->context->input;
   if (console->selectable && top->input_processed == nk_false &&
       widget_bounds.w > 0 && nk_input_is_mouse_moved(input) &&
       nk_input_is_mouse_hovering_rect(input, widget_bounds)) {
     nk_console_set_active_widget(console);
-    // Calculate the active child index based on where the mouse is within
-    // widget_bounds. This assumes each column has equal width.
-    float columnWidth = (float)widget_bounds.w / console->columns;
-    float mouseX = input->mouse.pos.x - widget_bounds.x;
-    row->data.activeChild = 0;
-    for (float x = 0; x < widget_bounds.w;) {
-      float right =
-          x + row->data.children[row->data.activeChild].columns * columnWidth;
-      if (right > mouseX) {
-        break;
-      }
-      x = right;
-      row->data.activeChild += 1;
+
+    // Find the child that the mouse is hovering over.
+    float x = (input->mouse.pos.x - widget_bounds.x) / widget_bounds.w;
```

Comment:
Good catch. Didn't think of that. Is there a widget that allows multiple spanning columns? Was considering that for the Spacebar in the textedit.

Reply:
For example check boxes or combo boxes both use 2 columns. As I said it'd be kind of weird to do this but the following is technically possible and mousing only works correctly with the loop approach:
```c
nk_console* row = nk_console_row_begin(console);
nk_console_button(row, "1");
nk_bool check = false;
nk_console_checkbox(row, "2", &check);
nk_console_button(row, "3");
nk_console_row_end(row);
```

Reply:
I see. And since this code currently sets all row child `columns` to `0`, it ends up forcing them all to appear in single columns. Unsure of a good solution so far. Do you have any ideas? Will this loop still work? Maybe a `nk_bool do_not_mess_with_row_stuff` property or something.

![Screenshot from 2024-06-26 03-19-02](https://github.com/RobLoach/nuklear_console/assets/25086/2a412c9d-f5d3-4c29-a35d-237d31da7db4)

Reply:
Let's continue this discussion in the thread itself. Thanks!